Kinder/UsersLikedItems.xaml.cs

```diff
@@ -141,5 +166,140 @@ private void UnlikeButton_Click(object sender, RoutedEventArgs e)
+                //read current liked
+                using (StreamReader reader = new StreamReader(FileLocation_liked))
+                {
+                    while (!reader.EndOfStream)
+                    {
+                        int[] tempArr = Temp.ParsedLiked(reader.ReadLine());
+                        List<int> tempList = new();
+
+                        for (int i = 1; i < tempArr.Length; i++)
+                        {
+                            tempList.Add(tempArr[i]);
+                        }
+
+                        TempList.Add(new LikedItemsClass(tempArr[0], tempList));
+                    }
+                }
+
+                //delete the absent ones
+                for (int i = 0; i < TempList.Count; i++)
+                {
+                    TempList[i].ItemsIDs.Remove(itemID);
+                }
+
+                //upload updated liked list to file
+                using (StreamWriter writer = new StreamWriter(FileLocation_liked))
+                {
+                    foreach (LikedItemsClass liked in TempList)
+                    {
+                        writer.WriteLine(liked.ToString());
+                    }
+                }
+
+                using (StreamWriter w = new(System.IO.Path.Combine(Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName, "Data_files\\Claimed.txt")))
+                {
+                    foreach (var item in GivenItems)
+                    {
+                        if(item.ItemID != itemID)
+                        {
+                            w.WriteLine(item.ItemID.ToString() + ';' + item.ID.ToString());
+                        }
+                    }
+                }
+            }
+        }
+
+        private void ReleaseKarmaPoints(Item item)
+        {
+            User tempUser = User.GetUserByID(item.UserID);
+
+            tempUser.KarmaPoints += item.KarmaPrice;
+
+            FileManager.ChangeUserField(tempUser);
+        }
+
+        private void ClaimButton_Click(object sender, RoutedEventArgs e)
+        {
+            Item classObj = ItemsTable.SelectedItem as Item;
+            Data temp = new Data(classObj.ID, User.CurrentUserID);
+
+            if (GivenItems.Contains(temp))
+            {
+                Items.Remove(classObj);
+                ItemsTable.Items.Clear();
+                LoadData();
+
+                ItemsList.Remove(classObj);
+                GivenItems.Remove(temp);
+                ReWriteFile(classObj.ID);
+
+                ReleaseKarmaPoints(classObj);
+            }
+            else
+            {
+                MessageBox.Show("This item is not given to you");
+            }
+        }
+
+        private class Data : User, IEquatable<Data>
+        { 
+            public int ItemID { get; set; }
+
+            public Data(int ItemID, int UserID)
+            {
+                this.ID = UserID;
+                this.ItemID = ItemID;
+            }
+
+            public Data()
+            {
+
```

Comment:
what is the purpose of this?

Reply:
sometimes we have to create empty Data type element in list